source/nodejs/adaptivecards-designer/src/card-designer-surface.ts

```diff
@@ -290,7 +291,12 @@ export class CardDesignerSurface {
         let cardToRender: Adaptive.AdaptiveCard = this.card;
 
         if (asPreview) {
-            let inputPayload = this.card.toJSON(this._serializationContext);
+
+            let inputPayload = this._currentPayload;
+
+            if (!this._currentPayload) {
+                inputPayload = this.card.toJSON(this._serializationContext);
```

Comment:
I'm wondering if you handle errors correctly or not. For example, an invalid json. we need to warning user which line and what's the error

Reply:
I believe rendering and parsing errors are handled within adaptivecards. Once the card has been parsed and rendered, we display the errors in the designer.

@licanhua Is that what you mean by warning the user?

Reply:
yes. Because there is no link to test it. I don't know there is error correctly showed or not